### PR TITLE
Channel Broker to manage channels on threads

### PR DESCRIPTION
**Pull request by biinari**

### CODE REVIEW

File: lib/hutch/broker.rb [hunk 1]
```diff
@@ -313,6 +321,17 @@ def ensure_connection!(routing_key, message)
       raise_publish_error('connection is closed', routing_key, message) unless @connection.open?
     end
 
+    def channel_broker
+      if Thread.current[CHANNEL_BROKER_KEY] && !Thread.current[CHANNEL_BROKER_KEY].active
+        Thread.current[CHANNEL_BROKER_KEY].reconnect(open_channel!)
+      end
+      Thread.current[CHANNEL_BROKER_KEY] ||= ChannelBroker.new(open_channel!)
```
Comment (evanrolfe) [inline lib/hutch/broker.rb]: I think it would be better to have this as an if/else statement.

**Comment by AMHOL** [comment]
:+1: all looks good to me

**Comment by brentmurphy** [comment]
:+1:

**Comment by biinari** [comment]
Hmm, need to set up the exchange within each channel, some more work to be done

File: lib/hutch/broker.rb (hunk 1 above)
Comment (biinari) [inline lib/hutch/broker.rb]: Not convinced that makes it any clearer @Evan, you'd then need to have a final `Thread.current[CHANNEL_BROKER_KEY]` after the if/else statement

Comment (evanrolfe) [inline lib/hutch/broker.rb]: fair point :+1:

Comment (AMHOL) [inline lib/hutch/broker.rb]: Could move it to a helper, `reconnect_if_inactive` or something

Comment (biinari) [inline lib/hutch/broker.rb]: Changed the code more anyway. The lines above no longer exist

**Comment by ednp** [comment]
:+1:

**Pull request merged**